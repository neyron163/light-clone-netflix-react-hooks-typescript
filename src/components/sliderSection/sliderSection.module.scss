.item {
  min-width: 258px;
  height: 145px;
  position: relative;
  margin-right: 6px;

  img {
    position: absolute;
    height: 100.5%;
    width: 100.5%;
    transform: translate(-50%, -50%);
    top: 50%;
    left: 50%;
  }
}

.sliderSection {
  margin: 60px 0 50px;

  &:last-child {
    margin-bottom: 0;
  }
}
