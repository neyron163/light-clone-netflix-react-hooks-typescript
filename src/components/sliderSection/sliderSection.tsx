import React, { ReactElement } from "react";
import Slider from "react-slick";
import { moviesTypes, settings } from "../../types";

import s from "./sliderSection.module.scss";

interface PropsType {
  settings: settings;
  data: Array<moviesTypes>;
}

export const SliderSection = ({ settings, data }: PropsType): ReactElement => (
  <div className={s.sliderSection}>
    <Slider {...settings}>
      {data.map(
        (element: moviesTypes): ReactElement => (
          <div className={s.item}>
            <img src={element.preview} alt={element.preview} />
          </div>
        )
      )}
    </Slider>
  </div>
);
