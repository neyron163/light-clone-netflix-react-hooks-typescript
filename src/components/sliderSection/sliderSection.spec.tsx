import React from "react";
import renderer from "react-test-renderer";

import { SliderSection } from "./sliderSection";

jest.mock("react-slick");

const props = {
  settings: {
    infinite: true,
    speed: 500,
    slidesToShow: 6,
    slidesToScroll: 7
  },
  data: [
    {
      title: "Friends",
      preview: "img"
    }
  ]
};

describe("<SliderSection />", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<SliderSection {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
