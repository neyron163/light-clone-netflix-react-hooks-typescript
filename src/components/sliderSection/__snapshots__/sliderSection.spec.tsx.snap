// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SliderSection /> should render correctly 1`] = `
<div
  className="sliderSection"
/>
`;
